### PR TITLE
Add ActionBar component

### PR DESCRIPTION
**Pull request by sscholle**
## Proposed changes

- add Action Bar component (with minor refactor)
- add tests
- add storybook stories
- added supplementary css styles and mixins

## Types of changes

What type of changes does your code introduce?
_Put an `x` in the boxes that apply_

- [x] New feature
- [ ] Bugfix
- [ ] Quality of Life

## Checklist

_Put an `x` in the boxes that apply._

- [ ] Linter passes (`yarn lint`)
- [ ] Unit Test pass (`yarn jest`)
- [ ] I have added at least two useful tests for new component (if applicable)

## Relevant Issues

- https://github.com/bonlineza/objectives-and-key-results/issues/18
- https://github.com/bonlineza/objectives-and-key-results/issues/17
- https://github.com/bonlineza/objectives-and-key-results/issues/2

### CODE REVIEW

File: src/components/ActionBar/index.js [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    }
+  };
+
+  componentDidUpdate = prevProps => {
+    if (this.props.messages.length !== prevProps.messages.length) {
+      this.show(this.props.visibleDuration);
+    }
+  };
+
+  componentWillUnmount = () => {
```
Comment (robguy21) [inline src/components/ActionBar/index.js]: I don't think we should leave in lifecycles that have been announced to be deprecated

File: src/components/ActionBar/scss/ActionBar.scss [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+.action-bar {
+  @include menu-anchor-collapse();
+  @include link--transition();
+  @include soft--sides;
+  height: 100%;
+  display: flex;
+  align-items: center;
+  justify-content: flex-start;
+  background-color: black;
```
Comment (robguy21) [inline src/components/ActionBar/scss/ActionBar.scss]: i don't think this should be a style you need to override

File: src/components/Map/scss/MapMarker.scss [hunk 3]
```diff
@@ -0,0 +1,10 @@
+/*------------------------------------*\
+    $MAP MARKER
+\*------------------------------------*/
+
+.map-marker {
+  width: 3rem;
+  height: 1rem;
+  background-color: black;
+  color: white;
```
Comment (robguy21) [inline src/components/Map/scss/MapMarker.scss]: why are we moving away from only keeping functional css rules?

File: src/components/ActionBar/index.js (hunk 1 above)
Comment (sscholle) [reply src/components/ActionBar/index.js]: true, but doesn't apply to this function

File: src/components/Map/scss/MapMarker.scss (hunk 3 above)
Comment (sscholle) [reply src/components/Map/scss/MapMarker.scss]: this should be removed...

File: src/components/ActionBar/scss/ActionBar.scss (hunk 2 above)
Comment (sscholle) [reply src/components/ActionBar/scss/ActionBar.scss]: sure, but then that could be said more or less for the others also.... where do we stop?

File: src/components/ActionBar/index.js (hunk 1 above)
Comment (robguy21) [reply src/components/ActionBar/index.js]: why does it not apply?

File: src/components/ActionBar/scss/ActionBar.scss (hunk 2 above)
Comment (robguy21) [reply src/components/ActionBar/scss/ActionBar.scss]: Similar to my other comment- this moves away from "functional" styles and into stylistic decisions.

**Pull request merged**